File: tests/record/reorder_fields/record_reorder_fields.cpp
typedef struct {
  int x;
  int y;
  int z;
  int m;
} obj;


class OBJ{
  int x;

  public:
    void f(int y) {
      x = y + x;
    }
    void g(int y){
      x = y - x;
    }

};

int main() {
  obj a = {0, 1, 2, 3};
  int x;
  x = 5;
  return 0;
}

File: tests/record/reorder_fields/record_reorder_fields.ml

open Optitrust
open Target
open Prelude


let _ = Run.script_cpp (fun _ ->

  !! Record_basic.reorder_fields (Move_before ("x", ["m";"z"])) [cTypDef "obj"];
  !! Record_basic.reorder_fields (Move_after ("y", ["z"])) [cTypDef "obj"];
  !! Record_basic.reorder_fields (Move_after ("z", ["y";"m"])) [cTypDef "obj"];
  !! Record_basic.reorder_fields (Reorder_all ["x";"y";"z"; "m"]) [cTypDef "obj"];

  !! Record_basic.reorder_fields (Move_before ("x", ["f";"g"])) [cTypDef "OBJ"];
  !! Record_basic.reorder_fields (Reorder_all ["x";"f";"g"]) [cTypDef "OBJ"];

)
